operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -261,27 +258,10 @@ public void start() throws OperatorException {
     failOnMissingCurrentNS();
 
     try {
-      // check that the custom resource is known by the cluster if configured that way
-      final CustomResourceDefinition crd; // todo: check proper CRD spec version based on config
-      if (ConfigurationServiceProvider.instance().checkCRDAndValidateLocalModel()
-          && CustomResource.class.isAssignableFrom(resClass)) {
-        crd = kubernetesClient.apiextensions().v1().customResourceDefinitions().withName(crdName)
-            .get();
-        if (crd == null) {
-          throwMissingCRDException(crdName, specVersion, controllerName);
-        }
-
-        // Apply validations that are not handled by fabric8
-        CustomResourceUtils.assertCustomResource(resClass, crd);
```

Comment:
I'm also fine to have this functionality, just target defatult feature switch to false.

Reply:
Yes, maybe we should keep the feature after all, though I have to admit that if it's off by default, I'm not sure people will ever use it… 🤔 
That said, it could be activated by default in the extensions for some context (e.g. Quarkus' dev mode).

Reply:
KK will issue an another PR, with default false.